compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -300,10 +307,28 @@ impl<'tcx> EvalCtxt<'tcx> {
     }
 }
 
-fn fixme_instantiate_canonical_query_response<'tcx>(
-    _: &InferCtxt<'tcx>,
-    _: &OriginalQueryValues<'tcx>,
-    _: CanonicalResponse<'tcx>,
+fn instantiate_canonical_query_response<'tcx>(
+    infcx: &InferCtxt<'tcx>,
+    original_values: &OriginalQueryValues<'tcx>,
+    response: CanonicalResponse<'tcx>,
 ) -> Certainty {
-    unimplemented!()
+    let Ok(rustc_infer::infer::InferOk { value, obligations }) = infcx
```

Comment:
same here, please import these types instead of using long paths

Reply:
Oh yeah, I had just used the fully qualified paths because this was a hack and I didn't want to deal with import conflicts as I rebased this PR around :laughing: